sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -178,150 +185,85 @@ export abstract class V4PositionManager {
     }
   }
 
-  public static addCallParameters(_position: Position, _options: AddLiquidityOptions): MethodParameters {
+  // TODO: Add Support for permit2 batch forwarding
+  public static addCallParameters(position: Position, options: AddLiquidityOptions): MethodParameters {
     /**
      * Cases:
      * - if pool does not exist yet, encode initializePool
      * then,
-     * - if is mint, encode MINT_POSITION and then SETTLE_PAIR
-     * - else, encode INCREASE_LIQUIDITY and then SETTLE_PAIR
+     * - if is mint, encode MINT_POSITION. If it is on a NATIVE pool, encode a SWEEP. Finally encode a SETTLE_PAIR
+     * - else, encode INCREASE_LIQUIDITY. If it is on a NATIVE pool, encode a SWEEP. Finally encode a SETTLE_PAIR
      */
-    const calldatas = ['0x', '0x']
-    let value = toHex(0)
+    invariant(JSBI.greaterThan(position.liquidity, ZERO), 'ZERO_LIQUIDITY')
 
-    return {
-      calldata: Multicall.encodeMulticall(calldatas),
-      value,
-    }
-  }
+    const calldatas: string[] = []
```

Comment:
nit: given that "data" is already plural, maybe we can name it "calldataSet" or "calldataList"

Reply:
yeahh thats true, I think this was a pattern we just use elsewhere but I agree with you so I changed it to calldataList